src/templates/PoolV2/Overview/index.tsx

```diff
@@ -40,7 +56,28 @@ const Overview = ({ pool, handleClickStakeButton }: IOverviewProps) => {
     Record<string, string>
   >({})
 
+  const [RebalancingProgress, setRebalancingProgress] =
+    React.useState<IRebancingProgressProps | null>(null)
+  const [rebalanceWeights, setRebalanceWeights] =
+    React.useState<IRebalanceWeightsProps>(null)
+  const [listTokenWeights, setlistTokenWeights] = React.useState<
+    IlistTokenWeightsProps[]
+  >([])
+  const [activeIndex, setActiveIndex] = React.useState(0)
+  const tokenSeleted = listTokenWeights[activeIndex] ?? {}
+  const allocationsDataChart = listTokenWeights.map(item => ({
+    image: item.token.logo,
+    symbol: item.token?.symbol || '',
+    value: Number(item.allocation)
+  }))
+  const router = useRouter()
+  const poolId = Array.isArray(router.query.address)
+    ? router.query.address[0]
+    : router.query.address ?? ''
```

Comment:
I guess this is impossible to happen but this might get an undefined behaviour because you're checking if its an array, what if its an empty array?

```suggestion
  const poolId = Array.isArray(router.query.address) && router.query.address[0]
    ? router.query.address[0]
    : router.query.address ?? ''
```